omnibus/config/projects/agent.rb

```diff
@@ -97,6 +97,20 @@
   end
 end
 
+if heroku_target?
+  force_build_and_packaging = true
+  generate_distro_package = true
+else
+  force_build_and_packaging = false
+  if ENV["OMNIBUS_PACKAGE_ARTIFACT_DIR"]
+    dependency "package-artifact"
+    generate_distro_package = true
+    disable_healthcheck true
+  else
+    generate_distro_package = false
+  end
+end
```

Comment:
I find this hard to read. We have two flags (`a` = we do the build, `b` = we do the packaging), represented by two booleans `force_build_and_packaging = a and b`, `generate_distro_package = b`. I'd have an easier time with one boolean for `a`, and one for `b`.

I think the section would be easier to read and maintain like this, WDYT?

```
build = false
package = false
if ENV["OMNIBUS_PACKAGE_ARTIFACT_DIR"]
  package = true
else
  build = true
end
# Special case for Heroku: we build and package in the same job
if heroku_target?
  build = true
  package = true
end

[...]

package :deb do
  skip_packager !package
  [...]
end

[...]

if build
  dependency 'datadog-agent'
  [...]
  dependency 'datadog-cf-finalize'
end
if package && !build
  disable_healthcheck true
  dependency "package-artifact"
end
if package
  dependency "init-scripts-agent"
end
```

Reply:
that's much clearer indeed, done locally, I'll update the PR shortly